analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/NullPointerDereferenceBase.cs

```diff
@@ -20,34 +20,45 @@
 
 using SonarAnalyzer.SymbolicExecution.Constraints;
 
-namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks;
+
+public abstract class NullPointerDereferenceBase : SymbolicRuleCheck
 {
-    public abstract class NullPointerDereferenceBase : SymbolicRuleCheck
-    {
-        internal const string DiagnosticId = "S2259";
+    internal const string DiagnosticId = "S2259";
 
-        protected virtual bool IsSupressed(SyntaxNode node) => false;
+    protected virtual bool IsSupressed(SyntaxNode node) => false;
 
-        protected override ProgramState PreProcessSimple(SymbolicContext context)
+    protected override ProgramState PreProcessSimple(SymbolicContext context)
+    {
+        if (ReferenceOrDefault(context.Operation.Instance) is { Syntax: var syntax} reference
+            && context.HasConstraint(reference, ObjectConstraint.Null)
+            && !IsSupressed(syntax)
+            && SemanticModel.GetTypeInfo(syntax).Nullability().FlowState != NullableFlowState.NotNull)
         {
-            var reference = context.Operation.Instance.Kind switch
-            {
-                OperationKindEx.Invocation => context.Operation.Instance.ToInvocation().Instance,
-                OperationKindEx.PropertyReference => context.Operation.Instance.ToPropertyReference().Instance,
-                OperationKindEx.Await => context.Operation.Instance.ToAwait().Operation,
-                OperationKindEx.ArrayElementReference => context.Operation.Instance.ToArrayElementReference().ArrayReference,
-                _ => null,
-            };
-            if (reference != null
-                && context.HasConstraint(reference, ObjectConstraint.Null)
-                && !reference.Type.IsStruct() // ToDo: IsStruct() is a workaround before MMF-2401
-                && !IsSupressed(reference.Syntax)
-                && SemanticModel.GetTypeInfo(reference.Syntax).Nullability().FlowState != NullableFlowState.NotNull)
-            {
-                ReportIssue(reference, reference.Syntax.ToString());
-            }
-
-            return context.State;
+            ReportIssue(reference, syntax.ToString());
         }
+        return context.State;
     }
+
+    private static IOperation ReferenceOrDefault(IOperation operation) =>
+        operation.Kind switch
+        {
+            OperationKindEx.Invocation => InvocationOrDefault(operation),
+            OperationKindEx.PropertyReference => PropertyReferenceOrDefault(operation),
+            OperationKindEx.Await => operation.ToAwait().Operation,
+            OperationKindEx.ArrayElementReference => operation.ToArrayElementReference().ArrayReference,
+            _ => null,
```

Comment:
@pavel-mikula-sonarsource I noticed, that this neither covers fields nor events. Is there a good reason for it? If not, should I create a new issue for it and add it to the sprint?